src/components/TxNotification.tsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import { abbreviateString } from 'utils'
+import { Network } from 'types'
+import { useWalletConnection } from 'hooks/useWalletConnection'
+
+interface TxNotificationProps {
+  txHash: string
+}
+
+function getDomainPrefix(networkId: Network): string {
+  return !networkId || networkId === Network.Mainnet ? '' : Network[networkId].toLowerCase() + '.'
+}
+
+export const TxNotification: React.FC<TxNotificationProps> = ({ txHash }: TxNotificationProps) => {
+  const { networkId } = useWalletConnection()
```

Comment:
This can be undefined/null

I would add a:
```js
if (!networkId){
  return null
}
```

...so it doesn't draw anything if there's no connection

Reply:
@DmitryBespalov  what is best returning null/undefined/empty_string or it doesn't matter?

Reply:
doesn't matter

Reply:
@anxolin so you rather not return anything in case the wallet is not connected?

Right, this is not likely to happen, but thinking in terms of interface, considering that we'll use this initially inside a toast notification, the notification will be completely empty.

Unless you mean to only display the notification if there is one, such as:

```
const notification = <TxNotification .../>
if (notification) {
  toast.info(notification...)
}
```

Is that what you have in mind?

Reply:
Yes I meant that, but, that's a good point. So then I would say that this would affect if a user:

1. Create a tx
2. Disconnects
3. Receives the confirmation

You are right, showing a empty notification in that case is a bid bad. The best I can come up with is to "remember" the last network id. In the described event, the user had previously a network and then disconnected. We can try to have always the last networkw we remember :)

Reply:
That (cache networkId) I believe makes sense to have in the `useWalletInfo` hook, so from that point of view, no changes in this component regarding to that.

I'll make the caching and return null from this component.

Reply:
Well, caching in the hook is not as trivial as I thought. The hook doesn't update the walletinfo itself, but instead pass the update funtion as a callback. If this is a must, we should deal with it in another issue/PR

Reply:
It's not a must 👍 
Good suggestion. I would leave it for now